x/sysproxy/sysproxy_linux.go

```diff
@@ -0,0 +1,76 @@
+// Copyright 2024 Jigsaw Operations LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build linux
+
+package sysproxy
+
+import (
+	"fmt"
+	"os/exec"
+)
+
+type ProxyType string
+
+const (
+	httpProxyType  ProxyType = "http"
+	httpsProxyType ProxyType = "https"
+	ftpProxyType   ProxyType = "ftp"
+)
+
+func SetProxy(ip string, port string) error {
+	// Execute Linux specific commands to set proxy
+	if err := setManualMode(); err != nil {
```

Comment:
You should enable manual mode after you set the proxies, so whoever observing this change can get the right values. Also in case one of them fails.